packages/ember-metal/lib/mixin.js

```diff
@@ -257,8 +257,12 @@ function mergeMixins(mixins, m, descs, values, base, keys) {
 
 let IS_BINDING = /^.+Binding$/;
 
-function detectBinding(obj, key, value, m) {
-  if (IS_BINDING.test(key)) {
+export function detectBinding(obj, key, value, m) {
+  if (key.length < 8) { return false; }
+
+  let lastIndex = key.indexOf('Binding', key.length - 7);
```

Comment:
@cibernox - Well, its a bit different.  If we used `lastIndexOf`, it would continue to search the string (in reverse) until it got to the first character.  In our case, we only want to check if the last 7 chars match exactly `Binding`, so we use `.indexOf('Binding', length - 7)` so it effectively only has to check a single possible match.